### ISSUE TITLE
数据加载流程疑问

### ISSUE THREAD

**Issue by qianlinjun** [TencentYoutuResearch/ObjectDetection-OneStageDet#64]
谢谢分享!
有个数据加载代码方面的问题, 在_voc_train.py文件中有下面一段代码:
dataset = VOCDataset(hyper_params)
dataloader = data.DataLoader(
            dataset,
            batch_size = self.mini_batch_size,
            shuffle = True,
            drop_last = True,
            num_workers = hyper_params.nworkers if self.cuda else 0,
            pin_memory = hyper_params.pin_mem if self.cuda else False,
            collate_fn = data.list_collate,
        )
这个dataset里的每一张图像的标注框是一个字典结构,我没有看到处理这个字典结构的代码,想问一下这个字典结构在哪里经过处理变成一个比如float型列表,然后送进网络计算的?

**Comment by skywalkerfmc** [comment]
> 谢谢分享！
> 有个数据加载代码方面的问题，在_voc_train.py文件中有下面一段代码：
> 数据集= VOCDataset（hyper_params）
> 的DataLoader = data.DataLoader（
> 数据集
> 的batch_size = self.mini_batch_size，
> 洗牌=真，
> drop_last =正确，
> num_workers = hyper_params.nworkers，如果self.cuda else 0，
> pin_memory = hyper_params.pin_mem，如果self.cuda else，False，
> collat​​e_fn = data.list_collat​​e，
> ）
> 这个数据集没有看到处理这个字典结构的代码，想问一下这个字典结构在哪里经过处理变成一个浮动类型列表，然后送进网络计算的？

数据预处理是在_dataset_brambox.py 中def __getitem__(self, index) 这个里面处理的
